lib/Plugin/GhcTags/ETags/Formatter.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE BangPatterns   #-}
+{-# LANGUAGE NamedFieldPuns #-}
+
+-- | Simple etags formatter. See <https://en.wikipedia.org/wiki/Ctags#Etags>
+--
+module Plugin.GhcTags.ETags.Formatter where
+
+import qualified Data.ByteString as BS
+import           Data.ByteString.Builder (Builder)
+import qualified Data.ByteString.Builder as BB
+import qualified Data.Text          as Text
+import qualified Data.Text.Encoding as Text
+
+import           Plugin.GhcTags.Tag
+import           Plugin.GhcTags.Utils (endOfLine)
+
+
+-- | A product of two monoids: 'Builder' and 'Sum'.
+--
+data BuilderWithSize = BuilderWithSize {
+    builder     :: Builder,
+    builderSize :: !Int
+  }
+
+instance Semigroup BuilderWithSize where
+    BuilderWithSize b0 s0 <> BuilderWithSize b1 s1 =
+      BuilderWithSize (b0 <> b1) (s0 + s1)
+
+instance Monoid BuilderWithSize where
+    mempty = BuilderWithSize mempty 0
+
+
+formatTag :: Tag -> BuilderWithSize
+formatTag Tag {tagName, tagAddr} =
+    case mLineNo of
+      Nothing     -> mempty
+      Just lineNo -> flip BuilderWithSize (tagSize lineNo) $
+        -- TODO: get access to the original line or pretty print original
+        -- declaration
+           tagNameBuilder
+        <> BB.charUtf8 '\DEL' -- or '\x7f'
+        <> tagNameBuilder
+        <> BB.charUtf8 '\SOH' -- or '\x01'
+        <> BB.intDec lineNo
+        <> BB.charUtf8 ','
+        <> BB.intDec (succ lineNo)
```

Comment:
This should be a byte offset from the source file (I know hasktags also does the same thing as here, but that's wrong).

Reply:
Thanks - indeed I wasn't sure if this is right.  I think I can only estimate it with line number and column number.

What emacs does when the offsets are wrong?

Reply:
If there are multiple tags with the same name in one file (e.g. two or more data types that have a field `name` with `DuplicateRecordFields`), emacs will jump only to the first one.

Reply:
> I can only estimate it with line number and column number.

Hmm, `hpm_src_files` from HsParsedModule doesn't have path to the file being parsed?

Reply:
Yes we can use `ModSummary` to get the file path.